File: android/versioned-abis/expoview-abi42_0_0/src/main/java/abi42_0_0/expo/modules/facedetector/ExpoFaceDetectorProvider.java
package abi42_0_0.expo.modules.facedetector;

import android.content.Context;

import abi42_0_0.org.unimodules.core.interfaces.InternalModule;

import java.util.Collections;
import java.util.List;

import abi42_0_0.expo.modules.interfaces.facedetector.FaceDetectorInterface;
import abi42_0_0.expo.modules.interfaces.facedetector.FaceDetectorProviderInterface;

public class ExpoFaceDetectorProvider implements FaceDetectorProviderInterface, InternalModule {
  @Override
  public List<Class> getExportedInterfaces() {
    return Collections.singletonList(FaceDetectorProviderInterface.class);
  }

  public FaceDetectorInterface createFaceDetectorWithContext(Context context) {
    return new ExpoFaceDetector(context);
  }
}

File: android/src/main/java/expo/modules/facedetector/ExpoFaceDetectorProvider.kt
package expo.modules.facedetector

import android.content.Context
import expo.modules.core.interfaces.InternalModule
import expo.modules.interfaces.facedetector.FaceDetectorProviderInterface

class ExpoFaceDetectorProvider : FaceDetectorProviderInterface, InternalModule {
  override fun getExportedInterfaces() = listOf(FaceDetectorProviderInterface::class.java)

  override fun createFaceDetectorWithContext(context: Context) = ExpoFaceDetector(context)
}
